File: 프로그래머스/lv3/43162. 네트워크/네트워크.js
function dfs(n, computers, visited, start) {
    const stack = [start]; //dfs의 시작점을 초기화
    
    while (stack.length > 0) {
        const curr = stack.pop();
        visited[curr] = true;
        
        for(let i = 0; i < n; i++) {
            if(visited[i] === false && computers[curr][i] === 1) {
                stack.push(i);
            }
        }
    }
}
function solution(n, computers) {
    let answer = 0;
    const visited = new Array(n).fill(false);
    for(let i = 0; i < n; i++) {
        if(visited[i] === false) {
            dfs(n, computers, visited, i);
            answer += 1;
        }
    }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [0] * n
    
    def dfs(com):
        visited[com] = 1
        for i in range(n):
            if i != com and visited[i] == 0 and computers[i][com] == 1:
                dfs(i)
    
    for com in range(n):
        if visited[com] == 0:
            dfs(com)
            answer += 1
    return answer